src/main/java/AWSClientFactory.java

```diff
@@ -14,40 +14,50 @@
 import com.amazonaws.ClientConfiguration;
 import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.auth.AWSCredentialsProvider;
 import com.amazonaws.services.codebuild.AWSCodeBuildClient;
 import com.amazonaws.services.codebuild.model.InvalidInputException;
 import com.amazonaws.services.logs.AWSLogsClient;
 import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.AmazonClientException;
+import lombok.Getter;
+import org.apache.commons.lang.StringUtils;
 
 public class AWSClientFactory {
 
-    private final String proxyHost;
-    private final String proxyPort;
-    private final String awsAccessKey;
-    private final String awsSecretKey;
+    @Getter private boolean isInstanceProfileCredentialUsed = true;
     private final String region;
-
     private ClientConfiguration clientConfig;
     private AWSCredentials awsCredentials;
 
     public AWSClientFactory(String proxyHost, String proxyPort, String awsAccessKey, String awsSecretKey, String region) throws InvalidInputException {
 
-        Validation.checkAWSClientFactoryConfig(proxyHost, proxyPort, awsAccessKey, awsSecretKey);
+        // Priority is IAM credential > Instance Profile Credential
+        try {
+            AWSCredentialsProvider cp = InstanceProfileCredentialsProvider.getInstance();
```

Comment:
I think it would be more useful to default to DefaultAWSCredentialsProviderChain, so users can specify creds in multiple ways outside of configuring them directly in the plugin:
http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html

Since this implements AWSCredentialsProvider, you would need to change the awsCredentials property type (currently is AWSCredentials) and use AWSStaticCredentialsProvider for creds provided to the plugin.